lib/aio/github.py

```diff
@@ -133,17 +135,11 @@ async def open_issue(self, repo: str, issue: JsonObject) -> None:
         await self.post(f'repos/{repo}/issues', issue)
 
     async def read_file(self, subject: Subject, filename: str) -> str | None:
-        async def read_once() -> str | None:
-            try:
-                async with self.session.get(self.content / subject.repo / subject.sha / filename) as response:
-                    logger.debug('response %r', response)
-                    return await response.text()
-            except aiohttp.ClientResponseError as exc:
-                if exc.status == 404:
-                    return None
-                raise
-
-        return await retry(read_once)
+        content = await self.get_obj(f'repos/{subject.repo}/contents/{filename}', {'ref': subject.sha})
+        try:
+            return base64.b64decode(str(content['content'])).decode().strip()
+        except binascii.Error:
```

Comment:
This could easily toss UnicodeError too: you're requesting a blob, so the thing you get could well be binary...

But ...